### PR TITLE
Coverage for rust agent as github action

### PR DESCRIPTION
**Pull request by Koncpa**
Implement github action which gather code coverage of PR against master branch. And after merge it gather code coverage again. Add helper script which gather code coverage from testing farm. And update e2e plan where run tasks for measuring code coverage.

Signed-off-by: Patrik Koncity <pkoncity@redhat.com>

### CODE REVIEW

**Comment by Koncpa** [comment]
Hi guys,
for proper function of gathering code coverage for merged PR is it mandatory to have 
TESTING_FARM_API_TOKEN to access to TF machines. It can be used same token which is used already for [keylime](https://github.com/keylime/keylime) repository. Or you can get new one via this [steps](https://docs.testing-farm.io/general/0.1/onboarding.html). After you get the token you need to add as action secrets named as TESTING_FARM_API_TOKEN. @THS-on @mpeters

File: scripts/download_packit_coverage.sh [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+# maximum duration of the task in seconds
+MAX_DURATION="${MAX_DURATION:-5400}"  # 90 minutes
+
+# delay in seconds before doing another URL read
+# should not be too short not to exceed GitHub API quota
+SLEEP_DELAY="${SLEEP_DELAY:-120}"
+
+# github user/project we are going to work with
+PROJECT="Koncpa/rust-keylime"
```
Comment (aplanas) [inline scripts/download_packit_coverage.sh:33]: leftover?

Comment (Koncpa) [reply scripts/download_packit_coverage.sh]: Yeah, thanks for reminder.

File: packit-ci.fmf [hunk 2]
```diff
@@ -14,14 +14,14 @@
   prepare:
    - how: shell
      script:
-      - ln -s $(pwd) /var/tmp/rust-keylime_sources
+#      - ln -s $(pwd) /var/tmp/rust-keylime_sources
```
Comment (aplanas) [inline packit-ci.fmf:17]: leftover?

File: packit-ci.fmf [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
       - systemctl disable --now dnf-makecache.service || true
       - systemctl disable --now dnf-makecache.timer || true
 
   discover:
     how: fmf
-    url: https://github.com/RedHat-SP-Security/keylime-tests
-    ref: main
+    url: https://github.com/Koncpa/keylime-tests
```
Comment (aplanas) [inline packit-ci.fmf:23]: I guess that this will be dropped once WIP is removed

File: scripts/download_packit_coverage.sh [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            else
+                echo "$ERROR_MSG" 1>&2
+            fi
+            sleep ${SLEEP_DELAY}
+            DURATION=$(( ${DURATION}+${SLEEP_DELAY} ))
+        fi
+    done
+
+    if [ ${DURATION} -ge ${MAX_DURATION} ]; then
+         echo "Error: Maximum job diration exceeded. Terminating" 1>&2
```
Comment (aplanas) [inline scripts/download_packit_coverage.sh:86]: "duration"

File: packit-ci.fmf (hunk 2 above)
Comment (Koncpa) [reply packit-ci.fmf:17]: No, it will be dropped after WIP is removed.

File: packit-ci.fmf (hunk 3 above)
Comment (Koncpa) [reply packit-ci.fmf:23]: Yeah..

**Comment by Koncpa** [comment]
Probably it will fails for fedora-rawhide, but it this case is not relevant, code coverage is measured only for f37 and cs9.

**Review by ansasaki — APPROVED** [review]
LGTM!

Failures in Fedora Rawhide are expected, due to:
https://github.com/tpm2-software/tpm2-tools/issues/3210
https://bugzilla.redhat.com/show_bug.cgi?id=2171376

**Pull request merged**